tests/RequestHeaderParserTest.php

```diff
@@ -447,6 +447,18 @@ public function testQueryParmetersWillBeSet()
         $this->assertEquals('this', $queryParams['test']);
     }
 
+    public function testUnixRemoteSocketWithoutPort()
+    {
+        // reactphp/socket Connect->parseAddress returns "tcp://NUL" for UDS
+        $parser = new RequestHeaderParser(null, 'tcp://' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '' . "\0" . '');
```

Comment:
Why these empty string concatenations?

Reply:
See https://bugs.php.net/bug.php?id=74556. Thats what PHP returns for stream address of UDS on OSX.

Reply:
@andig Thanks for the update!

Have you seen https://github.com/reactphp/socket/pull/100? IMO this should be handled by the Socket component, which is also where this address comes from (afaict). This way, we would likely expect a `null` address here. Does this make sense to you?

Reply:
Why do you want upvotes on an already fixed bug? But it's not what I asked. You write `"\0" . '' . "\0"` there instead of simply `"\0\0"`, why that?

Reply:
> You write "\0" . '' . "\0" there instead of simply "\0\0", why that?

That was copy/paste from `var_export`, will fix.